Test compilation error

Since this repository sounded promising, I clone it and tried running the tests, but unfortunately I get a compilation error with g++ 12.2 (commit 5666e8cd770fa1fc6afbc3c24a36aad418332e73):

```
$ make -C tests/ check
make: Entering directory 'frozen/tests'
g++ -O3 -Wall -std=c++14 -march=native -Wextra -W -Werror -Wshadow -fPIC -I../include  -c -o test_main.o test_main.cpp
g++ -O3 -Wall -std=c++14 -march=native -Wextra -W -Werror -Wshadow -fPIC -I../include  -c -o test_rand.o test_rand.cpp
g++ -O3 -Wall -std=c++14 -march=native -Wextra -W -Werror -Wshadow -fPIC -I../include  -c -o test_set.o test_set.cpp
g++ -O3 -Wall -std=c++14 -march=native -Wextra -W -Werror -Wshadow -fPIC -I../include  -c -o test_map.o test_map.cpp
g++ -O3 -Wall -std=c++14 -march=native -Wextra -W -Werror -Wshadow -fPIC -I../include  -c -o test_unordered_set.o test_unordered_set.cpp
test_unordered_set.cpp: In function ‘void C_A_T_C_H_T_E_S_T_15()’:
test_unordered_set.cpp:195:18: error: the type ‘const frozen::unordered_set<frozen::basic_string<char>, 3, frozen::elsa<void>, C_A_T_C_H_T_E_S_T_15()::<lambda(const frozen::string&, const auto:1&)> >’ of ‘constexpr’ variable ‘set’ is not literal
  195 |   constexpr auto set = frozen::make_unordered_set<frozen::string>(
      |                  ^~~
In file included from test_unordered_set.cpp:2:
../include/frozen/unordered_set.h:49:7: note: ‘frozen::unordered_set<frozen::basic_string<char>, 3, frozen::elsa<void>, C_A_T_C_H_T_E_S_T_15()::<lambda(const frozen::string&, const auto:1&)> >’ is not literal because:
   49 | class unordered_set {
      |       ^~~~~~~~~~~~~
../include/frozen/unordered_set.h:55:18: note:   non-static data member ‘frozen::unordered_set<frozen::basic_string<char>, 3, frozen::elsa<void>, C_A_T_C_H_T_E_S_T_15()::<lambda(const frozen::string&, const auto:1&)> >::equal_’ has non-literal type
   55 |   KeyEqual const equal_;
      |                  ^~~~~~
test_unordered_set.cpp:191:20: note: ‘C_A_T_C_H_T_E_S_T_15()::<lambda(const frozen::string&, const auto:1&)>’ is not literal because:
  191 |   const auto eq = [](const frozen::string& frozen, const auto& str) {
      |                    ^
cc1plus: note:   ‘C_A_T_C_H_T_E_S_T_15()::<lambda(const frozen::string&, const auto:1&)>’ is a closure type, which is only literal in C++17 and later
make: *** [<builtin>: test_unordered_set.o] Error 1

```

```
$ g++ --version
g++ (Ubuntu 12.2.0-3ubuntu1) 12.2.0

```

Maybe it is easy to fix?

Thanks in advance,

Best regards,

Alain Mosnier

--- a/tests/test_unordered_set.cpp
+++ b/tests/test_unordered_set.cpp
@@ -185,16 +185,19 @@ TEST_CASE("frozen::unordered_set heterogeneous lookup", "[unordered_set]") {
   REQUIRE(set.find(std::string{"two"}, frozen::elsa<std::string>{}, eq) != set.end());
 }
 
+struct eq {
+  template<class StrTy>
+  auto operator()(const frozen::string &frozen, const StrTy &str) const {
+    return frozen == frozen::string{str.data(), str.size()};
+  }
+};
+
 TEST_CASE("frozen::unordered_set heterogeneous container", "[unordered_set]") {
   using namespace frozen::string_literals;
 
-  const auto eq = [](const frozen::string& frozen, const auto& str) {
-    return frozen == frozen::string{str.data(), str.size()};
-  };
-
   constexpr auto set = frozen::make_unordered_set<frozen::string>(
           {"one"_s, "two"_s, "three"_s},
-          frozen::elsa<>{}, eq);
+          frozen::elsa<>{}, eq{});
 
   REQUIRE(set.find(std::string{"two"}) != set.end());
   REQUIRE(set.find(frozen::string{"two"}) != set.end());

